tests/e2e/ansible/start_docker_registry.yaml

```diff
@@ -12,24 +12,16 @@
     local_registry_port: 5000
     local_registry_name: local-registry
   tasks:
-    - name: Install pip3
+    - name: Install python3-docker and python3-requests
       package:
-        # TODO: this is ubuntu specific...
-        name: python3-pip
-        state: present
-      retries: 3
-      delay: 10
-    # The docker and requests pip packages are required by the docker_container ansible module itself.
-    # The requests package is pinned to a version less than 2.32 to avoid a bug
-    # at https://github.com/docker/docker-py/issues/3256
-    - name: Install docker and requests pip packages
-      pip:
         name:
-          - docker
-          - requests<2.32
+          # This is Ubuntu specific
+          - python3-docker
+          - python3-requests
```

Comment:
My concern here is that due to a [breaking change](https://github.com/docker/docker-py/issues/3256) in requests 2.32, we had to have a version less than that (but great than 2.28 IIRC?

What version is `python3-requests` installing on the 3 different versions of Ubuntu we are supporting and are we at risk of falling into incompatibility again if there are distro updates, or do we think it is safe to rely on canonical to ensure that doesn't happen?

Reply:
This is a very good question and I'd like to challenge that it doesn't work with something that's not greater than 2.28.

Let's take a look of what we have in our CIs:
* azure -> Ubuntu 20.04 | python3-requests 2.22.0 -> CI passes with this PR
* azure -> Ubuntu 22.04 | python3-requests 2.25.1 -> CI passes with this PR
* TDX -> Ubuntu 24.04 | python3-request 2.31.0 -> CI passes with this PR
* s390x -> ? -> CI passes with this PR

When using PIP, the first thing that I've faced was:
```
fatal: [localhost]: FAILED! => {"attempts": 3, "changed": false, "cmd": ["/usr/bin/python3", "-m", "pip.__main__", "install", "--user", "docker", "requests<2.32"], "msg": "\n:stderr: error: externally-managed-environment\n\n× This environment is externally managed\n╰─> To install Python packages system-wide, try apt install\n    python3-xyz, where xyz is the package you are trying to\n    install.\n    \n    If you wish to install a non-Debian-packaged Python package,\n    create a virtual environment using python3 -m venv path/to/venv.\n    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make\n    sure you have python3-full installed.\n    \n    If you wish to install a non-Debian packaged Python application,\n    it may be easiest to use pipx install xyz, which will manage a\n    virtual environment for you. Make sure you have pipx installed.\n    \n    See /usr/share/doc/python3.12/README.venv for more information.\n\nnote: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.\nhint: See PEP 668 for the detailed specification.\n"}
```

So, being totally honest, I'd rather deal with the decision later on on whether using pip and facing the risk to break the system packages, or dealing with an update that **may** break ubuntu (and then, it seems TDX would be the first one receiving such update and it'd fall on me / Intel to solve it).

Let me know if you're comfortable enough with this, Steve.

Reply:
Ok, it might be that pre-2.28 doesn't work the latest python-docker and the distros have old docker that works, so if the tests pass I'm okay to carry this for now.